network.go

```diff
@@ -438,11 +438,9 @@ func (n *network) updateSvcRecord(ep *endpoint, isAdd bool) {
 	var epList []*endpoint
 	n.WalkEndpoints(func(e Endpoint) bool {
 		cEp := e.(*endpoint)
-		cEp.Lock()
-		if cEp.container != nil {
+		if _, hasSandbox := cEp.getSandbox(); hasSandbox {
```

Comment:
Yes, I am recollecting now that I am making the changes.
Most of these methods are being invoked as a result of an operation on the Endpoint. Mostly from Endpoint.Join/Leave.

Need some more careful refactoring while moving the methods under sandbox.